lib/mpl_toolkits/mplot3d/axis3d.py

```diff
@@ -256,26 +306,57 @@ def _get_coord_info(self, renderer):
 
         return mins, maxs, centers, deltas, bounds_proj, highs
 
-    def _get_axis_line_edge_points(self, minmax, maxmin):
+    def _get_axis_line_edge_points(self, minmax, maxmin, position=None):
         """Get the edge points for the black bolded axis line."""
         # When changing vertical axis some of the axes has to be
         # moved to the other plane so it looks the same as if the z-axis
         # was the vertical axis.
-        mb = [minmax, maxmin]
+        mb = [minmax, maxmin]  # line from origin to nearest corner to camera
         mb_rev = mb[::-1]
         mm = [[mb, mb_rev, mb_rev], [mb_rev, mb_rev, mb], [mb, mb, mb]]
         mm = mm[self.axes._vertical_axis][self._axinfo["i"]]
 
         juggled = self._axinfo["juggled"]
-        edge_point_0 = mm[0].copy()
-        edge_point_0[juggled[0]] = mm[1][juggled[0]]
+        edge_point_0 = mm[0].copy()  # origin point
+
+        if ((position == 'lower' and mm[1][juggled[-1]] < mm[0][juggled[-1]])
+            or (position == 'upper'
+                and mm[1][juggled[-1]] > mm[0][juggled[-1]])):
+            edge_point_0[juggled[-1]] = mm[1][juggled[-1]]
+        else:
+            edge_point_0[juggled[0]] = mm[1][juggled[0]]
 
         edge_point_1 = edge_point_0.copy()
         edge_point_1[juggled[1]] = mm[1][juggled[1]]
 
         return edge_point_0, edge_point_1
 
-    def _get_tickdir(self):
+    def _get_all_axis_line_edge_points(self, minmax, maxmin, axis_position=None):
+        # Determine edge points for the axis lines
+        edgep1s = []
+        edgep2s = []
+        position = []
+        if axis_position in (None, 'default'):
+            edgep1, edgep2 = self._get_axis_line_edge_points(minmax, maxmin)
+            edgep1s = [edgep1]
+            edgep2s = [edgep2]
+            position = ['default']
+        else:
+            edgep1_l, edgep2_l = self._get_axis_line_edge_points(minmax, maxmin,
+                                                                 position='lower')
+            edgep1_u, edgep2_u = self._get_axis_line_edge_points(minmax, maxmin,
+                                                                 position='upper')
+            if axis_position in ('lower', 'both'):
+                edgep1s.append(edgep1_l)
+                edgep2s.append(edgep2_l)
+                position.append('lower')
+            if axis_position in ('upper', 'both'):
+                edgep1s.append(edgep1_u)
+                edgep2s.append(edgep2_u)
+                position.append('upper')
+        return edgep1s, edgep2s, position
+
+    def _get_tickdir(self, position=None):
```

Comment:
It's always called with a position, isn't it?
```suggestion
    def _get_tickdir(self, position):
```